### ISSUE TITLE
Worker stuck after miner restarted

### ISSUE THREAD

**Issue by shaodan** [filecoin-project/lotus#4979]
**Describe the bug**
Worker with running tasks is stuck after miner restarted

```
2020-11-23T14:08:23.320Z        WARN    stores  stores/local.go:324     error reporting storage health for ecf316ed-404f-4caa-96ed-921d315b36ef ({Stat:{Capacity:23040836501504 Available:4423039802588 Reserved:9857808937764} Err:}): RPC client error: sendRequest failed:
Post "http://172.168.0.220:2345/rpc/v0": dial tcp 172.168.0.220:2345: connect: connection refused
2020-11-23T14:08:28.629Z        ERROR   main    lotus-seal-worker/main.go:515   LOTUS-MINER CONNECTION LOST
2020-11-23T14:08:28.645Z        INFO    main    lotus-seal-worker/main.go:467   Redeclaring local storage
2020-11-23T14:08:28.705Z        INFO    main    lotus-seal-worker/main.go:483   Making sure no local tasks are running
```

Worker with no running tasks reconnect fine
```
2020-11-23T14:08:25.225Z        WARN    stores  stores/local.go:324     error reporting storage health for 7dbfb046-c8b3-4884-8283-35fad547b0a5 ({Stat:{Capacity:420827041792 Available:106283106304 Reserved:0} Err:}): RPC client error: sendRequest failed: Post "http://172.168.0.220:2345/rpc/v0": dial tcp 172.168.0.220:2345: connect: connection refused
2020-11-23T14:08:34.376Z        ERROR   main    lotus-seal-worker/main.go:515   LOTUS-MINER CONNECTION LOST
2020-11-23T14:08:34.376Z        INFO    main    lotus-seal-worker/main.go:467   Redeclaring local storage
2020-11-23T14:08:34.384Z        INFO    main    lotus-seal-worker/main.go:483   Making sure no local tasks are running
2020-11-23T14:08:34.400Z        INFO    main    lotus-seal-worker/main.go:506   Worker registered successfully, waiting for tasks
```

**Version (run `lotus version`):**
lotus-miner version 1.2.1+git.df66f48f
lotus-worker version 1.2.1+git.df66f48f

**Comment by arajasek** [comment]
Hmm, that's interesting. Does the worker connect fine once it finishes its ongoing task?

cc @magik6k for whether this is expected

**Comment by shaodan** [comment]
@arajasek C2 workers reconnect successfully after ongoing task finished.

But for PC workers, even if wait-quit & reconnect works well, it's not economical for me to wait all tasks finish, since PC workers run 10+ PC1 tasks and 1 PC2 task in parallel.

I think the best way to solve this is reconnect without waiting, but as Magik [mentioned](https://github.com/filecoin-project/lotus/blob/660ac26dd92c07d3e4443f4eccc41ce4ad322c6b/cmd/lotus-seal-worker/main.go#L492)

> // TODO: we could get rid of this, but that requires tracking resources for restarted tasks correctly

**Comment by rjan90** [comment]
Hey!

This issue seems outdated, and the scheduler-logic on newer version of Lotus has been greatly improved. If you experience this issue in a current release please open a new ticket.

Thanks again!
